test/include.js

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var test = require('tape');
+
+var Thrift = require('../thrift').Thrift;
+var path = require('path');
+
+test('loads a thrift file with imports synchronously', function t(assert) {
+    var mainThrift = makeThriftLoader('include-parent.thrift')();
+    var importedThrift = makeThriftLoader('include-child.thrift')();
+
+    var typeImportedByMainThrift = mainThrift
+        .types
+        .BatchGetResponse
+        .fieldsByName
+        .items
+        .valueType
+        .rw
+        .valueType;
```

Comment:
I'm almost certain the current behavior is actually correct since this happens with sets, maps and lists. With these three, we want to compare the value type of the element of the set, map or list and  not the valueType of the set/map/list itself